.github/workflows/create_release.yml

```diff
@@ -30,7 +31,22 @@ jobs:
 
       - name: Build (opt)
         run: |
+          # The ref resolves to "main" for latest and e.g. "v0.1.0" for tagged.
+          REF=${{ github.ref }}
+          REF=${REF#refs/*}
+          # For branches, this is e.g. 'heads/<branch_name>', so remove 'heads/'.
+          REF=${REF#heads/*}
+          # For tags, this is e.g. 'tags/<branch_name>', so remove 'tags/'.
+          REF=${REF#tags/*}
+          if [[ "$REF" == "main" ]]; then
```

Comment:
Maybe it's a little cleaner (and a little more platform independent) to find the build version in separate steps:

- if: "startsWith(github.ref, 'refs/tags/v')"
  run: echo "build_ver=${GITHUB_REF#refs/tags/v}" >> $GITHUB_ENV
- if: "!startsWith(github.ref, 'refs/tags/v')"
  run: echo "build_ver=${GITHUB_SHA}" >> $GITHUB_ENV

On Windows, the run part would have to be adjusted a bit.

And then in the build step:   -copt='-DCDC_BUILD_VERSION=${{ env.build_ver }}'